### ISSUE TITLE
[3.6 & V.1.17 (Spices)] Cinnamon Crash when hitting All Apps on certain themes

### ISSUE THREAD

**Issue by dominichayesferen** [NikoKrause/CinnVIIStarkMenu#45]
Themes Tested to have this crash:
- Win9X by Elbullazul
- Windows Se7en (No colour variant) by Elbullazul

So, basically, on these themes, that I've seen so far, in Cinnamon 3.6, hitting All Apps, or just Searching, results in an instant Cinnamon Crash.
XSession Errors:
```
Cjs-Message: JS LOG: About to start Cinnamon
St-Message: cogl npot texture sizes SUPPORTED
Cjs-Message: JS LOG: Cinnamon started at Sat Nov 18 2017 11:56:46 GMT+0000 (GMT)
Cjs-Message: JS WARNING: [/usr/share/cinnamon/applets/SW++@mohammad-sn/applet.js 113]: reference to undefined property Settings.BindingDirection.TWO_WAY
Cjs-Message: JS LOG: network applet: Cannot find connection for active (or connection cannot be read)

(cinnamon:16035): St-WARNING **: Ignoring length property that isn't a number at line 1613, col 13
openGL version 3.3 detected (GL3 Cogl Driver)
MetaSyncRing disabled: couldn't find required GL extensions, or the minimum safe openGL version was not met
Cjs-Message: JS LOG: Invalid network device type, is 14
Cjs-Message: JS LOG: network applet: Found connection for active

(cinnamon:16035): St-WARNING **: Ignoring length property that isn't a number at line 323, col 26

(cinnamon:16035): St-WARNING **: Ignoring length property that isn't a number at line 323, col 26

(cinnamon:16035): St-WARNING **: Ignoring length property that isn't a number at line 323, col 26

(cinnamon:16035): St-WARNING **: Couldn't parse family in font property

(cinnamon:16035): St-WARNING **: Couldn't parse family in font property

(cinnamon:16035): St-WARNING **: Couldn't parse family in font property

(cinnamon:16035): St-WARNING **: Ignoring length property that isn't a number at line 1613, col 13

(cinnamon:16035): St-WARNING **: Couldn't parse family in font property

(cinnamon:16035): St-WARNING **: Ignoring length property that isn't a number at line 1613, col 13

(cinnamon:16035): St-WARNING **: Couldn't parse family in font property

(cinnamon:16035): St-WARNING **: Ignoring length property that isn't a number at line 1613, col 13

(cinnamon:16035): St-WARNING **: Couldn't parse family in font property

(cinnamon:16035): St-WARNING **: Ignoring length property that isn't a number at line 1613, col 13

(cinnamon:16035): St-WARNING **: Couldn't parse family in font property

(cinnamon:16035): St-WARNING **: Ignoring length property that isn't a number at line 1613, col 13

(cinnamon:16035): St-WARNING **: Couldn't parse family in font property

(cinnamon:16035): St-WARNING **: Ignoring length property that isn't a number at line 323, col 26

(cinnamon:16035): St-WARNING **: Ignoring length property that isn't a number at line 323, col 26

(cinnamon:16035): St-WARNING **: Ignoring length property that isn't a number at line 1613, col 13

(cinnamon:16035): St-WARNING **: Couldn't parse family in font property

(cinnamon:16035): St-WARNING **: Ignoring length property that isn't a number at line 323, col 26

(cinnamon:16035): St-WARNING **: Couldn't parse family in font property

(cinnamon:16035): St-WARNING **: Couldn't parse family in font property

(cinnamon:16035): St-WARNING **: Couldn't parse family in font property

(cinnamon:16035): St-WARNING **: Did not find color property '-gradient-start'

(cinnamon:16035): St-WARNING **: Did not find color property '-gradient-end'

(cinnamon:16035): St-WARNING **: Did not find color property '-gradient-start'

(cinnamon:16035): St-WARNING **: Did not find color property '-gradient-end'

(cinnamon:16035): St-WARNING **: Did not find color property '-gradient-start'

(cinnamon:16035): St-WARNING **: Did not find color property '-gradient-end'

(cinnamon:16035): St-WARNING **: Ignoring length property that isn't a number at line 323, col 26

(cinnamon:16035): St-WARNING **: Couldn't parse family in font property

(cinnamon:16035): St-WARNING **: Couldn't parse family in font property

(cinnamon:16035): St-WARNING **: Couldn't parse family in font property
**
St:ERROR:st/st-theme.c:1109:_st_theme_resolve_url: assertion failed: (base_filename)
```

Glass.log (Not that is has anything of use in it anyway):
`info t=2017-11-18T11:56:46.140Z Cinnamon.AppSystem.get_default() started in 127 ms
info t=2017-11-18T11:56:46.148Z loading user theme: /usr/share/themes/Windows Se7en/cinnamon/cinnamon.css
info t=2017-11-18T11:56:46.173Z added icon directory: /usr/share/themes/Windows Se7en/cinnamon
info t=2017-11-18T11:56:46.319Z PlacesManager: Updating devices
info t=2017-11-18T11:56:46.363Z Loaded extension transparent-panels@germanfr in 5 ms
info t=2017-11-18T11:56:46.364Z loaded at Sat Nov 18 2017 11:56:46 GMT+0000 (GMT)
info t=2017-11-18T11:56:47.162Z Loaded applet CinnVIIStarkMenu@NikoKrause in 795 ms
info t=2017-11-18T11:56:47.479Z Role locked: panellauncher
info t=2017-11-18T11:56:47.598Z Loaded applet IcingTaskManager@json in 434 ms
info t=2017-11-18T11:56:47.627Z Loaded applet keyboard@cinnamon.org in 27 ms
info t=2017-11-18T11:56:47.643Z Loaded applet SW++@mohammad-sn in 14 ms
info t=2017-11-18T11:56:47.670Z Loaded applet user@cinnamon.org in 25 ms
info t=2017-11-18T11:56:47.706Z Loaded applet a11y@cinnamon.org in 34 ms
info t=2017-11-18T11:56:47.722Z Loaded applet inhibit@cinnamon.org in 14 ms
info t=2017-11-18T11:56:47.750Z Loaded applet ScreenShot+RecordDesktop@tech71 in 26 ms
info t=2017-11-18T11:56:47.776Z Loaded applet removable-drives@cinnamon.org in 24 ms
info t=2017-11-18T11:56:47.789Z Role locked: notifications
info t=2017-11-18T11:56:47.795Z Loaded applet notifications@cinnamon.org in 17 ms
info t=2017-11-18T11:56:47.812Z Loaded applet show-hide-applets@mohammad-sn in 15 ms
info t=2017-11-18T11:56:47.843Z Loaded applet weather@mockturtl in 28 ms
info t=2017-11-18T11:56:47.865Z Loaded applet MessagingMenuV3@blub in 21 ms
info t=2017-11-18T11:56:47.877Z Loaded applet systray@cinnamon.org in 10 ms
info t=2017-11-18T11:56:47.996Z Loaded applet network@cinnamon.org in 117 ms
info t=2017-11-18T11:56:48.056Z Loaded applet sound@cinnamon.org in 57 ms
info t=2017-11-18T11:56:48.081Z Loaded applet power@cinnamon.org in 23 ms
info t=2017-11-18T11:56:48.126Z Loaded applet calendar@cinnamon.org in 42 ms
info t=2017-11-18T11:56:48.126Z AppletManager.init() started in 1762 ms
info t=2017-11-18T11:56:48.130Z Cinnamon took 2179 ms to start
info t=2017-11-18T11:56:48.665Z Hiding indicator (role already handled): nm-applet
info t=2017-11-18T11:56:48.960Z Adding systray: mintupdate.py (24x24px)
info t=2017-11-18T11:56:50.467Z Resized mintupdate.py with normalized size (22x22px)

**Comment by dominichayesferen** [comment]
[Windows Se7en.tar.gz](https://github.com/NikoKrause/CinnVIIStarkMenu/files/1484555/Windows.Se7en.tar.gz)
Here's the theme itself, in case you want to check the theme itself
NOTE: It seems to be theme-centric, and I also didn't have this issue back in 3.4, with an older CinnVII, also, it isn't all the Elbullazul themes, either that do this issue...

**Comment by NikoKrause** [comment]
Hey,
thanks for the report. It's not related to StarkMenu only. Happens with the default menu as well.
It has been reported here: https://github.com/linuxmint/Cinnamon/issues/7021

But nobody was able to reproduce it on Linux Mint till now. Thought it was Arch related. But it happens with your theme on Mint as well. I'll add the info there, so this can be targeted before the final release.

**Comment by jaszhix** [comment]
Yep, crashing. [Stack trace](https://0bin.net/paste/tL9I0jJofJy8VL4Q#2H76-PahGjFO9chbGlViYk3T/Rq3Bl9sxMCjtVWn/VY).

The resolve_url error to me would suggest one of the image urls are incorrect in the CSS file. Not that it should be able to crash Cinnamon or anything, but that's where I would start trying to fix the theme.

**Comment by NikoKrause** [comment]
Culprit was found. Fix will be comming.

For the Se7en theme it's the scrollbar hhandle part, which causes the crash.

**Comment by dominichayesferen** [comment]
@Elbullazul , you've got this.

**Comment by Elbullazul** [comment]
@feren maybe fixed by tomorrow...

**Comment by Elbullazul** [comment]
@feren so I commented the only reference of hhandle in the cinnamon file, has the issue reappeared?

**Comment by dominichayesferen** [comment]
Hang on, just redownloading the theme, to check

**Comment by dominichayesferen** [comment]
@Elbullazul Oh, and also, Windows 9X suffers the same bug too

**Comment by dominichayesferen** [comment]
@Elbullazul @NikoKrause  I can still replicate the crash on the latest version of the Windows Se7en Theme (GitHub) (no update for Cinnamon in the Sylvia PPA yet, so still on 3.6.2, if that's also related to the problem at hand):
[Windows Se7en.tar.gz](https://github.com/NikoKrause/CinnVIIStarkMenu/files/1485430/Windows.Se7en.tar.gz)

**Comment by NikoKrause** [comment]
The fix has been merged and will be available on 3.6.4
